### ISSUE TITLE
set com port baudrate in serial_driver.py

### ISSUE THREAD

**Issue by tom-slick** [ManiacalLabs/BiblioPixel#66]
With the python default baudrate of 9600 I was getting terrible performance between a RPI and and uno.  If set to something higher (like 1,000,000 from the AllPixel firmware) it works great.  If set to anything but 9600 in my sketch it would never connect.  Could this be an option in the constructor?

**Comment by adammhaile** [comment]
Are you using the GPIO serial pins on the Pi or something? We don't set the
baud rate in serial drive because it's assumed you are using a USB serial
device which doesn't really have a baud rate and ignores anything you set
it to. It just runs at USB speed. Could certainly add an option to the
__init__ for serial driver. Won't be able to get to that for at least a
week, but I've got an incoming update to BiblioPixel and I'll make sure
it's in that release.

**Comment by tom-slick** [comment]
Thank for the quick reply.  I was using an Arduino Uno and a USB cable.  I was looking for a dead easy and repeatable way to add this to a community project I am working on.  I'll add it to your forum show and tell when we get the bugs worked out.    

I have a pull request ready, but it would probably be easier for you to add it to yours then to fix conflicts in the pull.

**Comment by adammhaile** [comment]
Ok... yeah, the usb to serial conversion on the Uno is not so hot. I assume
you just loaded the AllPixel firmware onto it? You'll be limited in data
speed with the Uno and would have to upgrade to using an ATMega32u4 to
improve.

I'll try to take a look in a few days as I'll be traveling for a bit. Send
the pull request along. Even if I don't use it directly it will show me
what you need.

**Comment by tom-slick** [comment]
Yes I used the AllPixel firmware, that is okay I hope?  It will still carry the appropriate licence files and credits. 

The project is meant to be low-cost, that was one of the main goals from the beginning.

Here you go.  
Added baudrate to constructor #67 
And again thank you

**Comment by adammhaile** [comment]
Of course! I'm happy to have someone actually use it in something else :) I
believe you are the first.
I was just trying to get a clear picture of your whole system.

**Comment by rec** [comment]
Oops, #67 didn't work.  We'll get it going for the dev release though.

**Comment by rec** [comment]
@adammhaile, this is in the dev branch now...?

**Comment by adammhaile** [comment]
Don't think so... I'll pull it in. Like the piGPIO thing I was kind of
waiting on what happened with our BP3 plans.

**Comment by Goofybud16** [comment]
Just some notes I'd like to include based on my own testing with a Lanmu Uno R3 (Based on the MEGA328P-PU @ 16MHz):

This testing was performed with PixelWeb and BiblioPixel, installed using `pip2 install BiblioPixel` and `pip2 install PixelWeb`, then manually modifying the serial_driver.py file to add a baud-rate.

Maximum Reliable Baud Rate : # of Driven WS2812B LEDS

115200 : 5
57600 : 11
38400 : 17
28800 : Wouldn't init
19200 : 35
14400 : Wouldn't Init
9600 : 60 (All of the LEDs I have)

If you have, for example, 6 LEDs and try to use a baud rate of 115200, you would get at most a couple frames before the communications failed. It appears that the MCU just cannot signal out the LEDs fast enough to catch the next message. If you are running at a higher clock speed, these numbers will probably increase.

**Comment by adammhaile** [comment]
The AllPixel firmware was not designed to be usable on the ATMega328p. It's
Serial stack is sub par and in no way ready for this kind of required data
throughput. I appreciate the data but am curious why you were trying this
in the first place?

**Comment by Goofybud16** [comment]
>The AllPixel firmware was not designed to be usable on the ATMega328p.

I know. I was providing data for anyone else who was going to attempt using an ATMega328P with the firmware and BiblioPixel. (Such as the original poster, who was also using an Arduino Uno)

>why you were trying this in the first place?

I purchased a WS2812B strip with the hope of controlling them through my Raspberry Pi. That didn't work, and I didn't want to drop $20 on an AllPixel (+ the $10 I already spent on the strip) just to play with the strip. I just went for the cheapest Uno clone with fast shipping, since if I end up not using or liking the LED strip, I can easily use the Arduino for other things.

As it stands now, a 328P based setup looks to be a decent alternative to the AllPixel if you only need to control a maximum of 10-15 LEDs.
